lib/ansible/modules/clustering/kubevirt/kubevirt_raw.py

```diff
@@ -223,23 +223,26 @@ def _create_stream(self, resource, namespace, wait_time):
         return w, stream
 
     def _read_stream(self, resource, watcher, stream):
-        for event in stream:
-            if event.get('object'):
-                entity = ResourceInstance(resource, event['object'])
-                metadata = entity.metadata
-                if entity.status.phase == 'Bound':
-                    annotations = metadata.annotations if \
-                        metadata.annotations else {}
-                    IMPORT_STATUS_KEY = 'cdi.kubevirt.io/storage.import.pod.phase'
-                    import_status = annotations.get(IMPORT_STATUS_KEY)
-                    labels = metadata.labels if metadata.labels else {}
-                    if (not self._use_cdi(annotations, labels) or
-                            import_status == 'Succeeded'):
-                        watcher.stop()
-                        return entity
-                    elif entity.status.phase == 'Failed':
-                        watcher.stop()
-                        self.fail_json(msg="Failed to import PersistentVolumeClaim")
+        try:
+            for event in stream:
+                if event.get('object'):
+                    entity = ResourceInstance(resource, event['object'])
+                    metadata = entity.metadata
+                    if entity.status.phase == 'Bound':
+                        annotations = metadata.annotations if \
+                            metadata.annotations else {}
+                        IMPORT_STATUS_KEY = 'cdi.kubevirt.io/storage.import.pod.phase'
+                        import_status = annotations.get(IMPORT_STATUS_KEY)
+                        labels = metadata.labels if metadata.labels else {}
+                        if (not self._use_cdi(annotations, labels) or
+                                import_status == 'Succeeded'):
+                            watcher.stop()
+                            return entity
+                        elif entity.status.phase == 'Failed':
+                            watcher.stop()
+                            self.fail_json(msg="Failed to import PersistentVolumeClaim")
+        except Exception as exc:
```

Comment:
What kind of exceptions we would get here?

Reply:
No clue. But the other class does this inside its _read_stream() and it looks like a good idea in general, since if we were to have any kinds of exceptions, this would create a mildly more informative error message. And improving error reporting is one of the open issues.

Reply:
Which class? Can we reuse?

Reply:
https://github.com/kubevirt/ansible-kubevirt-modules/blob/master/lib/ansible/modules/clustering/kubevirt/kubevirt_scale_vmirs.py#L211 and no, we can't. From what I've looked at the code, it does not make sense to try to separate stream handling into a helper class. It's better to just make sure the stream handling code in both classes behave similarly (hence this patch) and then write some universal testing code to test both classes with. (I'm referring to the Watch thing.)

Reply:
What I did in ovirt modules, and found quite good, is to have just one try, except per module and re-reaise with traceback, that way debugging is OK, and code is not verbose.

Reply:
@machacekondra Please point Mariusz to that code so we can do the same here

Reply:
Yup, that looks better. Just pushed a change making all the modules have exception handling that way.